org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/breakpoints/StandardJavaBreakpointEditor.java

```diff
@@ -87,18 +99,60 @@ protected Button createCheckButton(Composite parent, String text) {
 	 *            the parent composite
 	 */
 	protected void createTriggerPointButton(Composite parent) {
-		Composite composite = SWTFactory.createComposite(parent, parent.getFont(), 1, 1, 0, 0, 0);
+		Composite composite = SWTFactory.createComposite(parent, parent.getFont(), 2, 1, 0, 0, 0);
 		fTriggerPointButton = createCheckButton(composite, PropertyPageMessages.JavaBreakpointPage_12);
-
+		fTriggerPointButton.setEnabled(true);
 		fTriggerPointButton.setSelection(isTriggerPoint());
 		fTriggerPointButton.addSelectionListener(new SelectionAdapter() {
 			@Override
 			public void widgetSelected(SelectionEvent event) {
+				if (isTriggerPoint()) {
+					if (isTriggerPoint() && fResumeOnHit.isEnabled()) {
+						fResumeOnHit.setSelection(false);
+						fResumeOnHit.setEnabled(false);
+					} else {
+						fResumeOnHit.setEnabled(true);
+					}
+					if (!fSuspendThread.getSelection() && !fSuspendVM.getSelection()) {
```

Comment:
The blocks with conditions repeat mostly same actions in both if/else, common code is unrelated to the if/else should be moved out.